src/sp/it/pl/audio/playback/VlcPlayer.kt

```diff
@@ -164,25 +164,44 @@ class VlcPlayer: GeneralPlayer.Play {
 
     companion object: KLogging() {
 
-        private fun discoverVlc(location: File) = NativeDiscovery(computeVlcDiscoverer(location), DefaultWindowsNativeDiscoveryStrategy(), DefaultLinuxNativeDiscoveryStrategy(), DefaultMacNativeDiscoveryStrategy()).discover()
-
-        private fun computeVlcDiscoverer(location: File) = object: DefaultWindowsNativeDiscoveryStrategy() {
-            override fun onGetDirectoryNames(directoryNames: MutableList<String>) {
-                directoryNames.clear()
-                directoryNames += location.absolutePath
-            }
+        private fun discoverVlc(location: File): Boolean {
+            val loc = location.absolutePath
+            return NativeDiscovery(WindowsDiscoverer(loc), LinuxDiscoverer(loc), MacDiscoverer(loc)).discover()
         }
 
         private fun Item.uriAsVlcPath() = uri.toASCIIString().let {
             // vlcj bug
             // https://github.com/sghpjuikit/player/issues/40
             // https://github.com/caprica/vlcj/issues/173
             // https://github.com/caprica/vlcj/issues/424
-            if (it.startsWith("file:/") && it[6] != '/')
+            if (it.startsWith("file:/") && it[6]!='/')
                 it.replaceFirst("file:/", "file:///")
             else it
         }
 
+        class WindowsDiscoverer(vararg val locations: String): DefaultWindowsNativeDiscoveryStrategy() {
```

Comment:
I dont like the fact that the discoverers take String instead of File. Why is it this way? There is also Path if we want to pass in absolute path so the implementations do not have to each compute that, but I think signature `vararg val locations: File` is better I also wonder what is the use of vararg if we have no such use case.

Ultimately it makes no sense to pass in the files at all. WIndows or LinuxDIscoverer has nothing to do with OurOwnDiscoverer which should be its own Discoverer. This is clear from jsut repeating the same logic in ever discoverer implementation, pls use the old style to create a custom one that looks in the provided directory;

So I think we want to always apply two discoverer one for the Os we are on and one for our custom location.

Reply:
No it's not the same, every discoverer has different files it searches for and a different OS it supports. The way you implemented it previously it'd only detect in the app folder on Windows. Look into the implementations or trust me ^^

The parameters, idk, doesn't make much of a difference, I just wanted to only call `absolutePath` once since they just need the String. It can also take file if you want.

Reply:
This would also be much easier if the maintainer of vlcj accepts my PR https://github.com/caprica/vlcj/pull/647

Reply:
Well the problem is that the methods search for different files for each OS, and that I'd like to not re-implement all this in my code. Because if anything changes, the library should handle it, otherwise we might encounter weird issues.

https://github.com/caprica/vlcj/pull/612 for example will add more directories for Linux, which is very important since vlc is actually in one of these directories for me, so are you gonna keep up with vlcj and copying all the chanegs to it over? Probably not, that's why I propose this solution. Yes, it is ugly and has bad design, but it has _by far_ the least maintenance work attached to it.

Reply:
I think there is a misunderstanding. I believe my solution involves no maintenance at all. I will investigate and get back to you at the weekend. Th you for patience.